### PR TITLE
Fixes for Iron Bank (#76087)

### PR DESCRIPTION
**Pull request by pugnascotia**
Second attempt after reverting #76087.

This PR fixes a number of issues with the Iron Bank Docker context. Most notably it changes the generated `Dockerfile` in the build context to `COPY` the Elasticsearch archive, instead of downloading it, since Iron Bank images have a different process and will never be built on Docker Hub. This requires that the context task be wired to `assemble` instead of the build image task. The latter is only used for testing purposes.

### CODE REVIEW

**Comment by pugnascotia** [comment]
@elasticmachine update branch

File: distribution/docker/build.gradle [hunk 1]
```diff
@@ -303,8 +310,10 @@ void addBuildDockerImageTask(Architecture architecture, DockerBase base) {
       onlyIf { Architecture.current() == architecture }
     }
 
-  tasks.named("assemble").configure {
-    dependsOn(buildDockerImageTask)
+  if (base != DockerBase.IRON_BANK) {
```
Comment (mark-vieira) [inline distribution/docker/build.gradle:313]: Won't this break release manager since now running `assemble` won't produce this artifact?

Comment (pugnascotia) [reply distribution/docker/build.gradle:313]: Shouldn't do - release-manager only cares about the Docker context, not the image.

Comment (mark-vieira) [reply distribution/docker/build.gradle:313]: Oh I see. Got it.

**Review by mark-vieira — APPROVED** [review]
:+1:

**Comment by pugnascotia** [comment]
Backported to `7.x` in c7fce0845ed.

**Pull request merged**